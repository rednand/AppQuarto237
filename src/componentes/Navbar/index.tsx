import React, { useState } from 'react';
import {
  Navigation,
  List,
  ItemLink,
  NavbarDropdown,
  NavbarDropdownContent,
  ListDrop,
  ItemLinkDrop,
} from './styles';

const Navbar = () => {
  const [click, setClick] = useState(false);
  const handleClick = () => setClick(!click);

  return (
    <Navigation>
      <List>
        <li>
          <ItemLink onClick={handleClick} to="/">
            Home
          </ItemLink>
        </li>
        <li>
          <ItemLink onClick={handleClick} to="/cadastro">
            Informe um filme
          </ItemLink>
        </li>
        <li>
          <ItemLink onClick={handleClick} to="/filmes">
            Filmes
          </ItemLink>
        </li>
      </List>
      <NavbarDropdown>
        <span>Menu</span>
        <NavbarDropdownContent>
          <ListDrop>
            <li>
              <ItemLinkDrop onClick={handleClick} to="/">
                Home
              </ItemLinkDrop>
            </li>
            <li>
              <ItemLinkDrop onClick={handleClick} to="/cadastro">
                Informe um filme
              </ItemLinkDrop>
            </li>
            <li>
              <ItemLinkDrop onClick={handleClick} to="/blog">
                Filmes
              </ItemLinkDrop>
            </li>
          </ListDrop>
        </NavbarDropdownContent>
      </NavbarDropdown>
    </Navigation>
  );
};

export default Navbar;
