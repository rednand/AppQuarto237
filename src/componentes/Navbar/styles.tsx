import { NavLink } from "react-router-dom";
import styled from "styled-components";

const COLORS = {
  primaryDark: "var(--bg200)",
  primaryLight: "#d44d24",
};

export const Navigation = styled.nav`
  opacity: ${(props) => (props ? "1" : "0")};
  transition: width 0.8s, opacity 0.8s;
`;

export const List = styled.ul`
  list-style: none;
  text-align: center;
  display: inline-flex;

  @media (max-width: 400px) {
    display: none;
  }
`;

export const ItemLink = styled(NavLink)`
  background-color: var(--bg900);
  height: 3rem;
  float: left;
  border-color: var(--bg900);
  font-size: 1.5vw;
  font-weight: 900;
  text-decoration: none;
  color: ${COLORS.primaryDark};
  padding: 0.6rem 2rem;
  background-image: linear-gradient(
    120deg,
    transparent 0%,
    transparent 50%,
    var(--bg200) 50%
  );
  background-size: 240%;
  transition: all 0.4s;
  &:hover,
  &:active {
    background-position: 100%;
    color: ${COLORS.primaryLight};
    background-color: ${COLORS.primaryDark};
    border-color: var(--bg100);
    transform: translateY(0.5rem);
  }
`;

export const NavbarDropdown = styled.div`
  font-size: 4vw;
  font-weight: 900;
  background-color: var(--bg900);
  text-align:center ;
  font-size: 2rem;
  width: 100px;
  color: #ffffff;
  margin-left: 0;
  background-color: var(--bg900);
  height: 3rem;
  display: none;
  position: relative;
  &:hover {
    display: block;
    > div {
      display: block;
    }
  }

  @media (max-width: 400px) {
    display: block;
  }
`;

export const NavbarDropdownContent = styled.div`
  display: none;
  position: absolute;
  background: linear-gradient(5rad, var(--bg900), #d41d24, var(--bg900));
  min-width: 180%;
  padding: 12px 16px;
  z-index: 1;
`;

export const ListDrop = styled.ul`
  list-style: none;
  text-align: center;
  display: inline-block;
  width: 100%;
`;

export const ItemLinkDrop = styled(NavLink)`
  background-color: var(--bg900);
  margin: 2%;
  height: auto;
  width: 100%;
  float: left;
  font-size: 5vw;
  font-weight: 900;
  text-decoration: none;
  color: ${COLORS.primaryDark};
  padding: .6rem 2rem;
  background-image: linear-gradient(
    120deg,
    transparent 0%,
    transparent 50%,
    var(--bg200) 50%
  );
  background-size: 240%;
  transition: all 0.4s;
  &:hover,
  &:active {
    background-position: 100%;
    color: ${COLORS.primaryLight};
    background-color: ${COLORS.primaryDark};
    border-color: var(--bg100);
    transform: translateY(0.5rem);
  }
`;
