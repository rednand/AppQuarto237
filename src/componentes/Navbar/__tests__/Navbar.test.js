import React from 'react';
import { render, screen } from '@testing-library/react';
import Navbar from '..';
import { MemoryRouter } from 'react-router-dom';
import {
  toBeInTheDocument,
  toHaveLength,
  toHaveClass,
} from '@testing-library/jest-dom';

test('Deve renderizar um link para a página inicial', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );
  const linksPaginaInicial = screen.queryAllByText('Home');

  const linkPaginaInicial = linksPaginaInicial.find(
    (link) => link.getAttribute('href') === '/',
  );

  expect(linkPaginaInicial).toBeInTheDocument();
});

test('Deve renderizar uma lista de links', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );

  const listaLinks = screen.getAllByRole('link');
  expect(listaLinks).toHaveLength(3);
});

test('Não deve renderizar o link para Reviews', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );

  const linkReview = screen.queryByText('Reviews');
  expect(linkReview).not.toBeInTheDocument();
});

test('Deve renderizar uma lista de links com a classe link', () => {
  render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );

  const links = screen.getAllByRole('link');
  links.forEach((link) => expect(link).not.toHaveClass('link'));
  expect(links).toMatchSnapshot();
});
