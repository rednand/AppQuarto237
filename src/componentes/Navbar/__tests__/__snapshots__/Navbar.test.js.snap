// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar uma lista de links com a classe link 1`] = `
[
  <a
    aria-current="page"
    class="sc-eqUAAy CARKJ active"
    href="/"
  >
    Home
  </a>,
  <a
    class="sc-eqUAAy CARKJ"
    href="/cadastro"
  >
    Informe um filme
  </a>,
  <a
    class="sc-eqUAAy CARKJ"
    href="/filmes"
  >
    Filmes
  </a>,
]
`;
